src/components/WalletInit/RestoreWallet/RestoreWalletScreen.js

```diff
@@ -86,93 +83,67 @@ const errorsVisibleWhileWriting = (errors) => {
 }
 
 const RestoreWalletScreen = (
-  {
-    navigateToWalletCredentials,
-    intl,
-    phrase,
-    setPhrase,
-    translateInvalidPhraseError,
-    isKeyboardOpen,
-    route,
-  }: {intl: IntlShape} & Object /* TODO: type */,
+  {intl, route, navigation}: {intl: IntlShape, navigation: Navigation} & Object /* TODO: type */,
 ) => {
   const implId: WalletImplementationId = route.params.walletImplementationId
+  const [phrase, setPhrase] = React.useState(CONFIG.DEBUG.PREFILL_FORMS ? CONFIG.DEBUG.MNEMONIC3 : '')
+
   const walletConfig = getWalletConfigById(implId)
   const errors = validateRecoveryPhrase(phrase, walletConfig.MNEMONIC_LEN)
-  const visibleErrors = isKeyboardOpen
+  const hasErrors = Object.keys(errors).length > 0
+  const visibleErrors = useSelector(isKeyboardOpenSelector)
     ? errorsVisibleWhileWriting(errors.invalidPhrase || [])
     : errors.invalidPhrase || []
 
-  const errorText = visibleErrors.map((error) => translateInvalidPhraseError(error)).join(' ')
+  const errorText = visibleErrors.map((error) => translateInvalidPhraseError(intl, (error: any))).join(' ')
+
+  const navigateToWalletCredentials = () => {
+    if (hasErrors) return
+
+    navigation.navigate(WALLET_INIT_ROUTES.VERIFY_RESTORED_WALLET, {
+      phrase: cleanMnemonic(phrase),
+      networkId: route.params.networkId,
+      walletImplementationId: route.params.walletImplementationId,
+    })
+  }
 
   return (
     <SafeAreaView style={styles.safeAreaView}>
       <StatusBar type="dark" />
 
-      <ScrollView keyboardDismissMode="on-drag">
-        <View style={styles.container}>
-          <Text>
-            {intl.formatMessage(messages.instructions, {
-              mnemonicLength: walletConfig.MNEMONIC_LEN,
-            })}
-          </Text>
-          <ValidatedTextInput
-            multiline
-            numberOfLines={3}
-            style={styles.phrase}
-            value={phrase}
-            onChangeText={setPhrase}
-            placeholder={intl.formatMessage(messages.mnemonicInputLabel)}
-            blurOnSubmit
-            error={errorText}
-            autoCapitalize="none"
-            keyboardType="visible-password"
-            // hopefully this prevents keyboard from learning the mnemonic
-            autoCorrect={false}
```

Comment:
I'd suggest to don't remove the autoCorrect.

Reply:
`autoCorrect={false}` is the default for TextInput`

Reply:
That's why I like to prefix our local stuff with some identifier. TextInput is the same name from rn components.